arbos/programs/testcompile.go

```diff
@@ -0,0 +1,240 @@
+// Copyright 2024, Offchain Labs, Inc.
+// For license information, see https://github.com/OffchainLabs/nitro/blob/master/LICENSE
+
+//go:build !wasm
+// +build !wasm
+
+package programs
+
+// This file exists because cgo isn't allowed in tests
+
+/*
+#cgo CFLAGS: -g -Wall -I../../target/include/
+#include "arbitrator.h"
+
+typedef uint16_t u16;
+typedef uint32_t u32;
+typedef uint64_t u64;
+typedef size_t usize;
+
+void handleReqWrap(usize api, u32 req_type, RustSlice *data, u64 *out_cost, GoSliceData *out_result, GoSliceData *out_raw_data);
+*/
+import "C"
+import (
+	"fmt"
+	"os"
+	"runtime"
+
+	"github.com/wasmerio/wasmer-go/wasmer"
+)
+
+func isNativeArm() bool {
+	return runtime.GOOS == "linux" && runtime.GOARCH == "arm64"
+}
+
+func isNativeX86() bool {
+	return runtime.GOOS == "linux" && runtime.GOARCH == "amd64"
+}
+
+func testCompileArch(store bool) error {
```

Comment:
these tests should be converted to use LocalTargetName(), rawdb.TargetArm / X86, and if possible - also DefaultStylusTargetConfig.Arm/X86